src/org/sosy_lab/java_smt/example/ListAllSolversWithFeatures.java

```diff
@@ -0,0 +1,311 @@
+        info.add(solverInfo);
+      }
+    }
+    return info;
+  }
+
+  /**
+   * Checks for solver-name, version, theories and features and packs them into a String[] of length
+   * 4.
+   *
+   * @param solver to check for information. Taken from Solvers enum only.
+   * @return String[] of length 4 of the solver you entered. Array content as String in the
+   *     following order: SolverName, SolverVersion, SolverTheories, SolverFeatures. String[] length
+   *     0 if invalid solver.
+   */
+  private List<String> getSolverInformation(Solvers solver)
+      throws SolverException, InterruptedException {
+    List<String> info = new ArrayList<>();
+
+    Configuration config = Configuration.defaultConfiguration();
+    LogManager logger = LogManager.createNullLogManager();
+    ShutdownNotifier notifier = ShutdownNotifier.createDummy();
+
+    try (SolverContext context =
+        SolverContextFactory.createSolverContext(config, logger, notifier, solver)) {
+
+      info.add(solver.name());
+
+      info.add(context.getVersion());
+
+      info.add(getTheories(context));
+
+      info.add(getFeatures(context));
+    } catch (InvalidConfigurationException e) {
+      // Catches missing solvers
+    }
+
+    return info;
+  }
+
+  /**
+   * Checks the solver entered for features. Features checked: Optimization, UnsatCore,
+   * UnsatCoreWithAssumptions, Assumptions, AllSat, Interpolation
+   *
+   * @param context SolverContext you want to check for features.
+   * @return String with the features of the entered solver separated by a comma. Empty if none
+   *     available.
+   */
+  private String getFeatures(SolverContext context) throws SolverException, InterruptedException {
+    List<String> features = new ArrayList<>();
+
+    // Optimization: Will throw UnsupportedOperationException in creation of prover if not
+    // available.
+    try (OptimizationProverEnvironment prover =
+        context.newOptimizationProverEnvironment(ProverOptions.GENERATE_MODELS)) {
+      if (prover != null) {
+        features.add("Optimization");
+      }
+    } catch (UnsupportedOperationException e) {
+    }
+
+    // Interpolation: Will throw UnsupportedOperationException in creation of prover if not
+    // available.
+    try (InterpolatingProverEnvironment<?> prover =
+        context.newProverEnvironmentWithInterpolation(ProverOptions.GENERATE_MODELS)) {
+      if (prover != null) {
+        features.add("Interpolation");
+      }
+    } catch (UnsupportedOperationException e) {
+    }
+
+    // UnsatCore: throws UnsupportedOperationException if not available.
+    // TODO: check behavior with null argument properly
+    try (ProverEnvironment prover =
+        context.newProverEnvironment(ProverOptions.GENERATE_UNSAT_CORE)) {
+      if (prover.getUnsatCore() != null) {
+        features.add("UnsatCore");
+      }
+    } catch (UnsupportedOperationException e) {
+    } catch (Exception e) {
```

Comment:
Why is this needed?

Reply:
Because i wanted minimal examples to check for `UnsupportedOperationException` and didn't yet check how the method `getUnsatCore()` behaves on an empty stack. This is subject to change.

Reply:
`UnsatCore()` throws different Exceptions based on the solver used (native Exceptions). I added a comment.